test/corpus/types.txt

```diff
@@ -761,7 +753,40 @@ alias NamedTuple2 = NamedTuple("x": Int32, y?: String, Z: Char, %q(hi again): In
           (constant))
         (named_type
           name: (identifier)
-          (constant)))))
+          (constant))
+        (named_type
+          name: (identifier)
+          (constant))
+        (named_type
+          name: (identifier)
+          (constant))))))
+
+================================================================================
+named tuple types with string keys
+================================================================================
+# Not tagged with :language(crystal) because the Crystal parser doesn't represent
+# these keys as strings
```

Comment:
In this case shouldn't the tree sitter also reflect that?

Reply:
No, because we would lose the string syntax highlighting in `{"x": Int32}`

Reply:
So how does this work for strings symbols?

Reply:
Internally the names of the tuple type fields are just strings: https://github.com/crystal-lang/crystal/blob/master/src/compiler/crystal/syntax/ast.cr#L722

So tuple types like `{x: Int32}` or `{"x": Int32}` will both be represented by the compiler as having `NamedArgument` nodes with `node.name == "x"`

Reply:
Right – what I mean is how does that work with string symbols in the tree sitter, because to me that sounds like `:"foo"` is also represented as a string.

Reply:
There are several rule variants for symbols, but they all alias to `$.symbol`
https://github.com/crystal-lang-tools/tree-sitter-crystal/blob/393ad9305c1b668e88fe0dc0cf1b926576ab7a56/grammar.js#L756-L777